web/modules/custom/dept_migrate/src/MigrateSupport.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+namespace Drupal\dept_migrate;
+
+use Drupal\Core\Database\Database;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\group\Entity\Group;
+use Drupal\group\Entity\GroupContent;
+use Drupal\node\NodeInterface;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class to support migrations.
+ */
+class MigrateSupport {
+
+  /**
+   * Database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $dbconn;
+
+  /**
+   * Database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $d7conn;
+
+  /**
+   * Drupal\Core\Entity\EntityTypeManagerInterface definition.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Constructs a new instance of this object.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->dbconn = Database::getConnection('default', 'default');
+    $this->d7conn = Database::getConnection('default', 'migrate');
+  }
+
+  /**
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to sync domain/groups for.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function syncDomainsToGroups(EntityInterface $entity) {
+    if ($entity instanceof NodeInterface) {
+      $this->syncNodeDomainsToGroups($entity);
+    }
+    elseif ($entity instanceof UserInterface) {
+      $this->syncUserDomainsToGroups($entity);
+    }
+    else {
+      throw new \Exception('Unhandled entity type for ' . $entity->getEntityTypeId());
+    }
+  }
+
+  /**
+   * @param string $domain_machine_name
+   *   The machine name of the D7 domain.
+   * @return int
+   *   The group id (GID) of the group entity.
+   */
+  public function domainToGroupId(string $domain_machine_name) {
```

Comment:
Static map/lookup as to my knowledge we're not routinely adding new sites/domains.